packages/core/src/domain/configuration/transportConfiguration.ts

```diff
@@ -47,10 +47,16 @@ export function computeTransportConfiguration(
 
   if (buildEnv.buildMode === BuildMode.E2E_TEST) {
     const e2eEndpointBuilder = (placeholder: string) => ({ build: () => placeholder } as EndpointBuilder)
-    configuration.internalMonitoringEndpointBuilder = e2eEndpointBuilder('<<< E2E INTERNAL MONITORING ENDPOINT >>>')
-    configuration.logsEndpointBuilder = e2eEndpointBuilder('<<< E2E LOGS ENDPOINT >>>')
-    configuration.rumEndpointBuilder = e2eEndpointBuilder('<<< E2E RUM ENDPOINT >>>')
-    configuration.sessionReplayEndpointBuilder = e2eEndpointBuilder('<<< E2E SESSION REPLAY ENDPOINT >>>')
+
+    const e2eEndpointBuilders = {
+      logsEndpointBuilder: e2eEndpointBuilder('<<< E2E LOGS ENDPOINT >>>'),
+      rumEndpointBuilder: e2eEndpointBuilder('<<< E2E RUM ENDPOINT >>>'),
+      sessionReplayEndpointBuilder: e2eEndpointBuilder('<<< E2E SESSION REPLAY ENDPOINT >>>'),
+      internalMonitoringEndpointBuilder: e2eEndpointBuilder('<<< E2E INTERNAL MONITORING ENDPOINT >>>'),
+    }
+
+    intakeEndpoints = objectValues(e2eEndpointBuilders).map((builder) => builder.build())
+    objectEntries(e2eEndpointBuilders).forEach(([key, builder]) => (configuration[key] = builder))
```

Comment:
industrializing the override of the configuration variable that we have just created does not seem right to me.
Could we have something more natural like:
```
// compute endpointBuilders and intakeEndpoints depending on build mode
// compute replica configuration and add intakes
// create configuration with correct values
```

Reply:
True, it could be better. I'm gonna try it